sbt-dumpschema-plugin/src/main/scala/org/virtuslab/akkasaferserializer/writer/SchemaWriter.scala

```diff
@@ -0,0 +1,44 @@
+package org.virtuslab.akkasaferserializer.writer
+
+import better.files.File
+import io.bullet.borer.Json
+import org.virtuslab.akkasaferserializer.DumpSchemaOptions
+import org.virtuslab.akkasaferserializer.model.TypeDefinition
+
+import scala.collection.mutable
+
+class SchemaWriter(outputDirectory: File) extends Codecs {
+
+  def this(dumpSchemaOptions: DumpSchemaOptions) = {
+    this(File(dumpSchemaOptions.outputDir))
+  }
+
+  lazy val lastDump: Map[String, TypeDefinition] = {
+    outputDirectory
+      .createDirectoryIfNotExists(createParents = true)
+      .list(_.extension.contains(".json"))
+      .flatMap { file =>
+        Json.decode(file.byteArray).to[TypeDefinition].valueTry.toOption
+      }
+      .map(x => (x.name, x))
+      .toMap
+  }
+
+  private val dumped: mutable.Set[String] = mutable.Set()
+
+  def isUpToDate(name: String): Boolean = dumped(name)
+  def consumeTypeDefinition(typeDefinition: TypeDefinition): Unit = {
+    if (lastDump.get(typeDefinition.name).fold(true)(_ != typeDefinition)) {
```

Comment:
Nit: `fold` on `Option` is hard to parse for humans... pls replace with `map`/`getOrElse`